tasks/cephfs/filesystem.py

```diff
@@ -73,16 +73,38 @@ def get_pgs_per_fs_pool(self):
     def create(self):
         pgs_per_fs_pool = self.get_pgs_per_fs_pool()
 
-        self.admin_remote.run(args=['sudo', 'ceph', 'osd', 'pool', 'create', 'metadata', pgs_per_fs_pool.__str__()])
-        self.admin_remote.run(args=['sudo', 'ceph', 'osd', 'pool', 'create', 'data', pgs_per_fs_pool.__str__()])
-        self.admin_remote.run(args=['sudo', 'ceph', 'fs', 'new', 'default', 'metadata', 'data'])
+        self.mon_manager.raw_cluster_cmd('osd', 'pool', 'create', 'metadata', pgs_per_fs_pool.__str__())
+        self.mon_manager.raw_cluster_cmd('osd', 'pool', 'create', 'data', pgs_per_fs_pool.__str__())
+        self.mon_manager.raw_cluster_cmd('fs', 'new', 'default', 'metadata', 'data')
+
+    def exists(self):
+        """
+        Whether a filesystem is enabled at all
+        """
+        fs_list = json.loads(self.mon_manager.raw_cluster_cmd('fs', 'ls', '--format=json-pretty'))
+        return len(fs_list) > 0
+
+    def delete_all(self):
+        """
+        Remove all filesystems that exist, and any pools in use by them.
+        """
+        fs_list = json.loads(self.mon_manager.raw_cluster_cmd('fs', 'ls', '--format=json-pretty'))
+        for fs_data in fs_list:
+            self.mon_manager.raw_cluster_cmd('fs', 'rm', fs_data['name'], '--yes-i-really-mean-it')
+            self.mon_manager.raw_cluster_cmd('osd', 'pool', 'delete',
+                                             fs_data['metadata_pool'], fs_data['metadata_pool'],
+                                             '--yes-i-really-really-mean-it')
+            for data_pool in fs_data['data_pools']:
+                self.mon_manager.raw_cluster_cmd('osd', 'pool', 'delete',
+                                                 data_pool, data_pool,
+                                                 '--yes-i-really-really-mean-it')
 
     def delete(self):
```

Comment:
I don't think anybody else invokes delete(); we should probably just delete it so it doesn't get outdated and then pick up an unaware user.